File: api-ruby/samples/server/shared.rb
require 'sinatra/base'
require_relative 'config'
module SharedRoutes
  def self.registered(app)

    app.get '/' do
      send_file File.join(settings.public_folder, 'index.html')
    end  
    
    app.get '/redirect' do
      send_file File.join(settings.public_folder, 'redirect.html')
    end

    app.get '/providers' do
      ipAddress = params[:ipAddress]
      verificationProfileId = ENV['TRINSIC_VERIFICATION_PROFILE_ID']
      
      # Create recommendation info with IP addresses if provided
      recommendation_info = TrinsicApi::RecommendationInfo.new({
        ip_addresses: ipAddress ? [ipAddress] : nil
      })
      
      # Create the recommend request with required verification_profile_id
      req = TrinsicApi::RecommendRequest.new({
        verification_profile_id: verificationProfileId,
        recommendation_info: recommendation_info
      })

      result = TrinsicServices::NETWORK.recommend_providers({recommend_request: req})
      json result.to_body
    end

    app.post '/exchange-result' do
      body = JSON.parse(request.body.read)
  
      session_id = body['sessionId']
      access_key = body['resultsAccessKey']
  
      opts = {
        get_session_result_request: TrinsicApi::GetSessionResultRequest.new({
          results_access_key: access_key
        })
      }
  
      result = TrinsicServices::SESSIONS.get_session_result(session_id, opts)
      json result.to_body
    end

  end
end

File: api-typescript/samples/server/src/shared.ts
import { SessionsApi } from "@trinsic/api";
import { Request, Response, NextFunction } from "express";
import type { Application } from "express"; // ✅ import the type explicitly

import path from "path";
import fs from "fs";
// Middleware to serve files if they exist
function serveStaticFileIfExists(staticDir: string) {
  return (req: Request, res: Response, next: NextFunction) => {
    let requestPath = req.path;
    if (req.path === "/") {
      requestPath = "/index.html"; // default to index.html
    }
    const filePath = path.join(staticDir, requestPath);
    fs.stat(filePath, (err, stats) => {
      if (!err && stats.isFile()) {
        res.sendFile(filePath);
      } else {
        fs.stat(path.join(staticDir, requestPath + ".html"), (err, stats) => {
          if (!err && stats.isFile()) {
            res.sendFile(path.join(staticDir, requestPath + ".html"));
          } else {
            next();
          }
        });
      }
    });
  };
}

export function sharedRoutes(
  app: Application,
  sessionsApi: SessionsApi
) {
  // Use the middleware for a specific base path
  app.use(
    "/",
    serveStaticFileIfExists(
      path.join(__dirname, "../../../../ui-web/samples/dist")
    )
  );

  app.get("/providers", async (req: any, res: any) => {
    const ipAddress = req.query.ipAddress;
    const result = await sessionsApi.recommendProviders({
      verificationProfileId: process.env.TRINSIC_VERIFICATION_PROFILE_ID!,
      recommendationInfo: {
        ipAddresses: ipAddress ? [ipAddress] : null,
      },
    });
    res.send(result);
  });

  app.post("/exchange-result", async (req: any, res: any) => {
    const result = await sessionsApi.getSessionResult(req.body.sessionId, {
      resultsAccessKey: req.body.resultsAccessKey,
    });
    res.send(result);
  });
}
